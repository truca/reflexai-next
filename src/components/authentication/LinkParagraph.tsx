import Link from "next/link";

interface LinkParagraphProps {
  paragraphText: string;
  linkText: string;
  linkHref: string;
}

export default function LinkParagraph({
  paragraphText,
  linkText,
  linkHref,
}: LinkParagraphProps) {
  return (
    <p className="mt-8">
      {paragraphText}{" "}
      <Link
        href={linkHref}
        className="text-blue-500 hover:text-blue-700 font-semibold"
      >
        {linkText}
      </Link>
    </p>
  );
}
