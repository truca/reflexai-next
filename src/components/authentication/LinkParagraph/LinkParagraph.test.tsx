import LinkParagraph from "./index";
import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";

describe("LinkParagraph", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <LinkParagraph
          linkHref="/login"
          paragraphText="paragraphText"
          linkText="linkText"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
