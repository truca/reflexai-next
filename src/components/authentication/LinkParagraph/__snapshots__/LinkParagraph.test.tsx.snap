// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkParagraph renders correctly 1`] = `
<p
  className="mt-8"
>
  paragraphText
   
  <a
    className="text-blue-500 hover:text-blue-700 font-semibold"
    href="/login"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    linkText
  </a>
</p>
`;
